src/chart/pie/labelLayout.ts

```diff
@@ -89,7 +89,7 @@ function adjustSingleSide(
             const rA = r + item.len;
             const rA2 = rA * rA;
             // Use ellipse implicit function to calculate x
-            const dx = Math.sqrt((1 - Math.abs(dy * dy / rB2)) * rA2);
+            const dx = Math.sqrt(Math.abs((1 - Math.abs(dy * dy / rB2)) * rA2));
```

Comment:
`const dx = Math.sqrt(Math.abs((1 - dy * dy / rB2) * rA2));` should be fine because `dy * dy` and `rB2` are always non-negative.